core/src/test/java/org/apache/iceberg/util/TestSortOrderUtil.java

```diff
@@ -303,4 +305,35 @@ public void testSortOrderClusteringWithRedundantPartitionFieldsMissing() {
         expected,
         SortOrderUtil.buildSortOrder(table.schema(), updatedSpec, order));
   }
+
+  @Test
+  public void testDropColumnFromDefaultSortOrder() {
+    TestTables.TestTable table =
+        TestTables.create(
+            tableDir, "test", SCHEMA, PartitionSpec.unpartitioned(), SortOrder.unsorted(), 1);
+    table.replaceSortOrder().asc("ts").commit();
+
+    AssertHelpers.assertThrows(
```

Comment:
nit: I've been trying to steer people away from using `AssertHelpers` (https://github.com/apache/iceberg/pull/6977), so this could be rewritten as
```
Assertions.assertThatThrownBy(() -> table.updateSchema().deleteColumn("ts").commit())
        .isInstanceOf(ValidationException.class)
        .hasMessage("Cannot find source column for sort field: identity(3) ASC NULLS FIRST");
```

Reply:
Ah, I just picked what was used in other tests as well. Great that you're working on removing them and have one way of doing it.